db/builder.cc

```diff
@@ -254,6 +258,25 @@ Status BuildTable(
       meta->marked_for_compaction = builder->NeedCompact();
       assert(meta->fd.GetFileSize() > 0);
       tp = builder->GetTableProperties(); // refresh now that builder is finished
+      if (memtable_payload_bytes != nullptr &&
+          memtable_garbage_bytes != nullptr) {
+        const CompactionIterationStats& ci_stats = c_iter.iter_stats();
+        uint64_t total_payload_bytes = ci_stats.total_input_raw_key_bytes +
+                                       ci_stats.total_input_raw_value_bytes +
+                                       total_tombstone_payload_bytes;
+        uint64_t total_payload_bytes_written =
+            (tp.raw_key_size + tp.raw_value_size);
+        // Prevent underflow, which may still happen at this point
+        // since we only support inserts, deletes, and deleteRanges.
+        if (total_payload_bytes_written < total_payload_bytes) {
```

Comment:
Can there be case  where memtable_garbage_bytes = 0 and memtable_payload_bytes is nonzero? Will that be covered?

Reply:
Oh this is a great catch! I am changing the "<" for a "<=". And yes this is typically the case for example for purely sequential write workload (no overwrites, just writes). in that situation, total_payload_bytes_written==total_payload_bytes, which must give you memtable_garbage_bytes = 0 and memtable_payload_bytes>0.